src/tfvc/repository.ts

```diff
@@ -43,9 +47,12 @@ export class Repository {
             new Status(ignoreFiles === undefined ? true : ignoreFiles));
     }
 
-    public async Version(): Promise<string> {
-        return this.RunCommand<string>(
-            new GetVersion());
+    public async CheckVersion(): Promise<void> {
+        if (!this._versionAlreadyChecked) {
+            this._versionAlreadyChecked = true;
```

Comment:
So, CheckVersion will throw the error if the version isn't correct. And I don't want to throw it twice, so I need to set the boolean before it might throw. I will add a comment.